specs/protocol/holocene/exec-engine.md

```diff
@@ -59,6 +71,29 @@ places a burden on users of the system in a post-fault-proofs world, where:
 Placing the [`L2ToL1MessagePasser`][l2-to-l1-mp] account storage root in the `withdrawalsRoot` field alleviates this burden
 for users and protocol participants alike, allowing them to propose and verify other proposals with lower operating costs.
 
+#### Genesis Block
+
+If Holocene is active at genesis block, the withdrawals root is the empty withdrawals root, regardless of L2 state.
+
+#### State Processing
+
+At the time of state processing, the header for which transactions are being validated should not make it's `withdrawalsRoot`
+available to the EVM/application layer.
+
+#### P2P
+
+During sync, the block body withdrawal hash is computed from the withdrawals list in the block body. If Holocene is active
+at such a block's header, we expect that the computed withdrawal hash must match the hash of an empty list of withdrawals.
+The header `withdrawalsRoot` hash can be any non-null value in this case. Verification is performed after a header is available
+so that the header's timestamp can be checked to see if Holocene is active.
+
+#### Backwards Compatibility Considerations
+
+Beginning at Shanghai and prior to Holocene activation, the `withdrawalsRoot` field is set to an empty withdrawals list.
+This is the same root as an empty storage root. The withdrawals are captured in the L2 state, however they are not
+reflected in the `withdrawalsRoot`. Any implementation that caclulates output root should be careful not to use the
+header `withdrawalsRoot`.
```

Comment:
Should be careful to only use the withdrawals root after holocene is active? which can be determined if its not empty

Reply:
If the smart contract storage is empty, it matches the empty withdrawals list, even though it represents something different pre-holocene.

And pre-holocene, even though the withdrawals root is present, and a valid MPT root in the block header, it shouldn't be used.

And post-holocene, if there was never any withdrawal contract storage, a MPT root equal to one of an empty list, can be valid to use.

Not sure how to best word a warning for these possible easy things to mistake, but I think it's worth mentioning something to this extent.